Confus/Confus/Player.cpp

```diff
@@ -2,52 +2,68 @@
 #include "Audio\PlayerAudioEmitter.h"
 #include "Player.h"
 #include "EventManager.h"
+#include "Flag.h"
 
 namespace Confus
 {
     const irr::u32 Player::WeaponJointIndex = 14u;
     const unsigned Player::LightAttackDamage = 10u;
     const unsigned Player::HeavyAttackDamage = 30u;
-    Player::Player(irr::IrrlichtDevice* a_Device)
-        : m_Weapon(a_Device->getSceneManager(), irr::core::vector3df(1.0f, 1.0f, 4.0f))
+    Player::Player(irr::IrrlichtDevice* a_Device, irr::s32 a_id, ETeamIdentifier a_TeamIdentifier)
+        : m_Weapon(a_Device->getSceneManager(), irr::core::vector3df(1.0f, 1.0f, 4.0f)),
+        irr::scene::ISceneNode(nullptr, a_Device->getSceneManager(), a_id),
+        TeamIdentifier(new ETeamIdentifier(a_TeamIdentifier)),
+        CarryingFlag(new EFlagEnum(EFlagEnum::None))
```

Comment:
Both of the above are leaking as well. Did you actually add this or is this from Christian's refactor? (it's in the player class, but I don't see this in Christian's pull req)